File: 06a.pl
#! /usr/bin/perl
use warnings;
use strict;
use feature qw{ say };

use ARGV::OrDATA;

INPUT:
while (<>) {
    for my $pos (3 .. 1 + length) {
        my $string = substr $_, $pos - 3, 4;
        if ($string !~ /(.).*\1/) {
            say $pos + 1;
            next INPUT
        }
    }
}

__DATA__
mjqjpqmgbljsphdztnvjfqwrcgsmlb
bvwbjplbgvbhsrlpgdmjqwftvncz
nppdvjthqldpwncqszvftbrmjlhg
nznrnfrfntjfmvfwmzdfjlvtqnbhcprsg
zcfzfwzzqfrljwzlrfnpqdbhtmscgvjw

File: 06a.py
#! /usr/bin/python3
import sys
import warnings

LENGTH = 4

with open(sys.argv[1], 'r') as fh:
    for line in fh:
        pos = LENGTH
        while pos < len(line):
            s = set(line[pos - LENGTH:pos])
            if len(s) == LENGTH:
                print(pos)
                break
            pos += 1
        else:
            warnings.warn('NOT FOUND')
